src/librustc/traits/select.rs

```diff
@@ -1671,6 +1671,19 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 self.assemble_builtin_bound_candidates(clone_conditions, &mut candidates)?;
             }
 
+            if lang_items.unpin_trait() == Some(def_id) {
+                // OK to skip binder because the substs on generator types never
+                // touch bound regions, they just capture the in-scope
+                // type/region parameters
```

Comment:
Updated comment.

That's what I was thinking of initially, but I couldn't find anything similar to that (if you have some pointers for where to look I can take another look at it). This definitely [gives the correct error message](https://github.com/rust-lang/rust/pull/55704/files#diff-d745a2cbea2137958fc6da09c5e6006aR1).

Reply:
Actually, I think I've found a better place to put this. In `assemble_candidates_from_auto_impls` I can just suppress adding a candidate for static generators when the trait is `Unpin`. I'll check whether that works now.

Reply:
@nikomatsakis @arielb1 Where would the correct place to put this be? Or does it not matter since the user can't write impls since generators are unnameable?